plugins/woocommerce/src/Internal/Traits/OrderAttributionMeta.php

```diff
@@ -0,0 +1,378 @@
+		} else {
+			return __( 'Desktop', 'woocommerce' );
+		}
+	}
+
+	/**
+	 * Set the meta fields and the field prefix.
+	 *
+	 * @return void
+	 */
+	private function set_fields_and_prefix() {
+		/**
+		 * Filter the fields to show in the source data metabox.
+		 *
+		 * @since x.x.x
+		 *
+		 * @param string[] $fields The fields to show.
+		 */
+		$this->fields = (array) apply_filters( 'wc_order_attribution_tracking_fields', $this->default_fields );
+		$this->set_field_prefix();
+	}
+
+	/**
+	 * Set the meta prefix for our fields.
+	 *
+	 * @return void
+	 */
+	private function set_field_prefix(): void {
+		/**
+		 * Filter the prefix for the meta fields.
+		 *
+		 * @since x.x.x
+		 *
+		 * @param string $prefix The prefix for the meta fields.
+		 */
+		$prefix = (string) apply_filters(
+			'wc_order_attribution_tracking_field_prefix',
+			'wc_order_attribution_'
+		);
+
+		// Remove leading and trailing underscores.
+		$prefix = trim( $prefix, '_' );
+
+		// Ensure the prefix ends with _, and set the prefix.
+		$this->field_prefix = "{$prefix}_";
+	}
+
+	/**
+	 * Filter the meta data to only the keys that we care about.
+	 *
+	 * @param WC_Meta_Data[] $meta The meta data.
+	 *
+	 * @return array
+	 */
+	private function filter_meta_data( array $meta ): array {
+		$return = array();
+		$prefix = $this->get_meta_prefixed_field( '' );
+
+		foreach ( $meta as $item ) {
+			if ( str_starts_with( $item->key, $prefix ) ) {
+				$return[ $this->unprefix_meta_field( $item->key ) ] = $item->value;
+			}
+		}
+
+		// Determine the device type from the user agent.
+		if ( ! array_key_exists( 'device_type', $return ) && array_key_exists( 'user_agent', $return ) ) {
+			$return['device_type'] = $this->get_device_type( $return );
+		}
+
+		// Determine the origin based on source type and referrer.
+		$source_type = $return['type'] ?? '';
+		switch ( $source_type ) {
+			case 'organic':
+				$origin = __( 'Organic search', 'woocommerce' );
+				break;
+
+			case 'referral':
+				$origin = __( 'Referral', 'woocommerce' );
+				break;
+
+			case 'typein':
+				$origin = __( 'Direct', 'woocommerce' );
+				break;
+
+			default:
+				$origin = __( 'Unknown', 'woocommerce' );
+				break;
+		}
+
+		$return['origin'] = $origin;
+
+		return $return;
+	}
+
+	/**
+	 * Get the field name with the appropriate prefix.
+	 *
+	 * @param string $field Field name.
+	 *
+	 * @return string The prefixed field name.
+	 */
+	private function get_prefixed_field( $field ): string {
+		return "{$this->field_prefix}{$field}";
+	}
+
+	/**
+	 * Get the field name with the meta prefix.
+	 *
+	 * @param string $field The field name.
+	 *
+	 * @return string The prefixed field name.
+	 */
+	private function get_meta_prefixed_field( string $field ): string {
+		// Map some of the fields to the correct meta name.
+		if ( 'type' === $field ) {
+			$field = 'source_type';
+		} elseif ( 'url' === $field ) {
+			$field = 'referrer';
+		}
+
+		return "_{$this->get_prefixed_field( $field )}";
+	}
```

Comment:
Is this behavior really needed? Can we just use `source_type` and `referrer` in the first place in (`OrderAttributionMeta::$default_fields`) ?